playground/mocks/data/idp/idx/error-authenticator-webauthn-failure.json

```diff
@@ -0,0 +1,299 @@
+        "value":[
+          {
+            "name":"authenticator",
+            "type":"object",
+            "options":[
+              {
+                "label":"Email",
+                "value":{
+                  "form":{
+                    "value":[
+                      {
+                        "name":"id",
+                        "required":true,
+                        "value":"aut12un5yKLBOCgqD0g4",
+                        "mutable":false
+                      },
+                      {
+                        "name":"methodType",
+                        "required":false,
+                        "value":"email",
+                        "mutable":false
+                      }
+                    ]
+                  }
+                },
+                "relatesTo":"$.authenticatorEnrollments.value[0]"
+              },
+              {
+                "label":"MacBook Touch ID",
+                "value":{
+                  "form":{
+                    "value":[
+                      {
+                        "name":"id",
+                        "required":true,
+                        "value":"aut12uqz8PnlDpzQ20g4",
+                        "mutable":false
+                      },
+                      {
+                        "name":"methodType",
+                        "required":false,
+                        "value":"webauthn",
+                        "mutable":false
+                      }
+                    ]
+                  }
+                },
+                "relatesTo":"$.authenticators.value[1]"
+              },
+              {
+                "label":"Password",
+                "value":{
+                  "form":{
+                    "value":[
+                      {
+                        "name":"id",
+                        "required":true,
+                        "value":"aut12umag8NYVaHJS0g4",
+                        "mutable":false
+                      },
+                      {
+                        "name":"methodType",
+                        "required":false,
+                        "value":"password",
+                        "mutable":false
+                      }
+                    ]
+                  }
+                },
+                "relatesTo":"$.authenticatorEnrollments.value[2]"
+              }
+            ]
+          },
+          {
+            "name":"stateHandle",
+            "required":true,
+            "value":"02krMfFqKXrnQTgs9qbm4BmG6dfTXR_g_tUR4kJXL5",
+            "visible":false,
+            "mutable":false
+          }
+        ],
+        "accepts":"application/json; okta-version=1.0.0"
+      }
+    ]
+  },
+  "messages":{
+    "type":"array",
+    "value":[
+      {
+        "message":"None of the WebAuthn authenticators the user has enrolled for was able to validate the assertion",
+        "i18n":{
+          "key":"authfactor.webauthn.error.assertion_validation_failure"
```

Comment:
I'm not a fan of this approach, as we're returning keys that will always be unused by clients.

What is the issue with returning the `authfactor.webauthn.error` key here? Developers can still read from the `message` value to understand what the root error is.

Reply:
Yes, it helps the developers to look in to the response if there is any error occur

Reply:
if backend is adding this key for this sole purpose of client to use its own translation  then it makes sense for backend to return the generic key. I am not sure if there is some other purpose for backend. 

Also seems like it is a clients decision to determine what message to show so it feels weird from backend to have a common error code. /cc @mpeng-okta

Reply:
And if we change the error object on the back-end now, it will cause to many effort, since webauthn probably the only one have this issue.

Reply:
Yeah, the full key is returned basically as a pipeline-level mechanism. When the IDX pipeline internationalizes keys, it does a translation/conversion for the message value itself, and returns the corresponding i18n key as well for the UI to grab and do its own translation for display in the SIW (the message value isn't necessarily displayed in the SIW AFAIK).

Would probably require some strange rewiring to have the pipeline do i18n on the full key (e.g. the assertion validation failure key above), but then only return the genericized i18n key, since the platform "layer" is using the full key to obtain the specific message above.